File: CivitasJava/src/Civitas/Sorpresa.java
package Civitas;

import java.util.ArrayList;

public abstract class Sorpresa {
    protected String texto;
    //private int valor;
   // private Tablero tablero;
    //private TipoSorpresa tipo;
    //private MazoSorpresas mazo;
    
    
    Sorpresa(String texto){
        this.texto = texto;
    }
    
    void informe(int actual, ArrayList<Jugador> todos){
        Diario.getInstance().ocurreEvento("Aplicando sorpresa: \n" + this.toString() + 
                " al jugador " + todos.get(actual).getNombre() + " de tipo: " + this.getClass());
    }
    
    abstract void aplicarAJugador(int actual, ArrayList<Jugador> todos);

    
    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos) {
        boolean es_correcto = false;
        if (actual >= 0 && actual < todos.size()){
            es_correcto = true;
        }
        return es_correcto;
    }
        
}

File: CivitasRuby/lib/sorpresa.rb
# encoding:utf-8

require_relative 'mazo_sorpresas'
require_relative 'tablero'
require_relative 'jugador'
require_relative 'diario'

module Civitas                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
  class Sorpresa    
    def initialize(texto)
      @texto = texto
    end
  
    attr_writer :texto
    
    def jugador_correcto(actual, todos)
      es_correcto = false
      if actual >= 0 && actual < todos.size
        es_correcto = true
      end
      return es_correcto
    end
        
  end
end
